Library/Homebrew/os/linux/ld.rb

```diff
@@ -7,13 +7,35 @@ module Linux
     module Ld
       sig { returns(String) }
       def self.brewed_ld_so_diagnostics
+        @brewed_ld_so_diagnostics ||= T.let(nil, T.nilable(String))
+        @retried_brewed_ld_so_diagnostics ||= T.let(false, T::Boolean)
+
+        # Check for `#nil?` here because @brewed_ld_so_diagnostics could be a blank string.
+        return @brewed_ld_so_diagnostics unless @brewed_ld_so_diagnostics.nil?
+
         brewed_ld_so = HOMEBREW_PREFIX/"lib/ld.so"
-        return "" unless brewed_ld_so.exist?
+        unless brewed_ld_so.exist?
+          @brewed_ld_so_diagnostics = ""
+          return @brewed_ld_so_diagnostics
+        end
```

Comment:
Does this behave correctly on fresh installs where this might not exist initially but does after the first dependency install?

`ld.so` should normally exist (it's created in the "global post-install") so doesn't seem like we want to cache the no-result.

Reply:
Haven't checked a fresh install, but not caching it seems the right move.